janitor/functions.py

```diff
@@ -1599,3 +1605,297 @@ def check(varname: str, value, expected_types: list):
                 varname=varname, expected_types=expected_types
             )
         )
+
+
+def _clean_accounting_column(x):
+    """
+    This function performs the logic for the `type == "accounting"`
+    attribute in make_currency_column_numeric.
+
+    It is intended to be used in a pandas `apply` method.
+    """
+
+    y = x.strip()
+    y = y.replace(",", "")
+    y = y.replace(")", "")
+    y = y.replace("(", "-")
+    if y == "-":
+        return 0.00
+    return float(y)
+
+
+def _make_currency_column_numeric(x, cast_non_numeric=None):
+    """
+    This function performs the logic for the changing cell values in
+    the make_currency_column_numeric fucntion.
+
+    It is intended to be used in a pandas `apply` method, after being passed
+    through `partial`.
+    """
+    acceptable_currency_characters = {
+        "-",
+        ".",
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "6",
+        "7",
+        "8",
+        "9",
+        "0",
+    }
+    if len(x) == 0:
+        return "ORIGINAL_NA"
+
+    if cast_non_numeric:
+        if x in cast_non_numeric.keys():
+            check(
+                "{%r: %r}" % (x, str(cast_non_numeric[x])),
+                cast_non_numeric[x],
+                [int, float],
+            )
+            return cast_non_numeric[x]
+        else:
+            return "".join(i for i in x if i in acceptable_currency_characters)
+    else:
+        return "".join(i for i in x if i in acceptable_currency_characters)
+
+
+def _replace_empty_string_with_none(x):
+    if isinstance(x, int):
+        return x
+
+    elif isinstance(x, float):
+        return x
+
+    elif len(x):
+        return x
+
+
+def _replace_original_empty_string_with_none(x):
+    if x != "ORIGINAL_NA":
+        return x
+
+
+@pf.register_dataframe_method
+def make_currency_column_numeric(
+    df,
+    col_name: str,
+    type: str = None,
+    cast_non_numeric: dict = None,
+    fill_all_non_numeric: float = None,
+    remove_non_numeric: bool = False,
+):
+    """
+    :param df: The DataFrame
```

Comment:
Python docstrings usually should have a one-liner summary of what it is doing.

In pyjanitor, the rest of the functions also describe in words what is being accomplished.

Could you add those two elements please?

Reply:
Whoops; added all the examples but not the summary!